plugins/wallabag.koplugin/main.lua

```diff
@@ -430,6 +430,12 @@ function Wallabag:getArticleList()
                           .. filtering
         local articles_json = self:callAPI("GET", articles_url, nil, "", "", true)
 
+        if articles_json == false then
```

Comment:
This logic can't be right. ;-) This is identical to replacing the `break` a few lines down with `return`.

I haven't had the time to look at it, but the most typical idiom is probably to pass a reason along.

For example, at its simplest:
```lua
false, "connection error"
```

While if it's an HTTP error code, maybe

```lua
false, "http", status
```

In any case, the problem is that a "proper" error is conflated with another type. Conflating them the other way around would retain that problem.

Reply:
Thanks, I didn't see initially that non-200 http responses were also returning false instead of nil. Errors are handled when retrieving entries now. Error handling for minor actions (adding tags, adding/removing articles) is not implemented yet.